tests/main/views/test_direct_award.py

```diff
@@ -1228,3 +1240,305 @@ def test_direct_award_outcome_scenarios(
                 "result": "awarded",
             }
+
+
+class TestDirectAwardOutcomeNonAwarded(DirectAwardSetupAndTeardown):
+    @pytest.mark.parametrize(
+        "endpoint_rel_path",
+        ("cancel", "none-suitable",)
+    )
+    def test_nonexistent_project_id(self, endpoint_rel_path):
+        res = self.client.post(
+            urljoin("/direct-award/projects/952738338097614/", endpoint_rel_path),
+            data=json.dumps(self._updated_by_data()),
+            content_type="application/json",
+        )
+        assert res.status_code == 404
+        assert json.loads(res.get_data()) == {
+            "error": "Project 952738338097614 not found",
+        }
+
+    # tuples of paramaters in the order:
+    #     has_active_search
+    #     has_non_active_search
+    #     project_locked
+    #     existing_outcome_result
+    #     existing_outcome_complete
+    #     expected_status_code
+    #     expected_response_data
+    _base_direct_award_nonawarded_scenarios = (
+        # "happy" paths
+        (True, True, True, None, False, 200, _anydict,),
+        (True, False, True, None, False, 200, _anydict,),
+        (False, True, True, "cancelled", False, 200, _anydict,),
+        (True, False, True, "awarded", False, 200, _anydict,),
+        (True, True, True, "cancelled", False, 200, _anydict,),
+        (True, False, False, "none-suitable", False, 200, _anydict),
+        (False, False, False, None, False, 200, _anydict),
+        # "failure" paths
+        (
+            True,
+            True,
+            True,
+            "cancelled",
+            True,
+            410,
+            {
+                "error": AnyStringMatching(r"Project \d+ already has a completed outcome: \d+"),
+            },
+        ),
+        (
+            True,
+            False,
+            True,
+            "awarded",
+            True,
+            410,
+            {
+                "error": AnyStringMatching(r"Project \d+ already has a completed outcome: \d+"),
+            },
+        ),
+    )
+
+    @pytest.mark.parametrize(
+        (
+            "endpoint_rel_path",
+            "has_active_search",
+            "has_non_active_search",
+            "project_locked",
+            "existing_outcome_result",
+            "existing_outcome_complete",
+            "expected_status_code",
+            "expected_response_data",
+        ),
+        tuple(chain(
+            (
+                # we test all of the _base_direct_award_nonawarded_scenarios against the known valid nonawarded reasons
+                (endpoint_rel_path,) + base_scenario
+                for endpoint_rel_path, base_scenario in product(
+                    ("none-suitable", "cancel",),
+                    _base_direct_award_nonawarded_scenarios,
+                )
+            ),
+            (
+                # but we also include a couple of specific cases with invalid reasons
+                (
+                    "severe-chill",
```

Comment:
I don't quite get why we're testing specifically two cases here

Reply:
To assert that it's not just an implementation that's taking `<project_id>/<any_string>` and happily trying to use that as a `result`.

Reply:
But one test of some arbitrary string would confirm that wouldn't it?

Reply:
I think I originally meant to test one of them as `awarded` but that idea got lost in the fog of my mind - will set it right...